### ISSUE TITLE
Regression with 24.08 : Hio module does not support Monochrome EXR images

### ISSUE THREAD

**Issue by seando-adsk** [PixarAnimationStudios/USD#3289]
### Description of Issue
We have a unit test in MayaUSD that reads OpenEXR images with 1, 2, and 4 channels in both half and float format to make sure every image type loads correctly.

Initially logged as issue #2800 which was fixed in **24.03/24.05**. However the issue is back again in **24.08**.

Looking at recent changes, replacing `strIsRed()` with a combined `strIsComponent()` suddenly lost the capability to decode a Y channel as being a potential red channel. Don't know if this is sufficient to cause the error we see though. Change is [here](https://github.com/PixarAnimationStudios/OpenUSD/commit/d50b6d13f6a75f2b0512e23c9dce0531b5e640ce#diff-b1cfc1ba8aa58d326de3654a75cc971e859e169847e19da750886bf6b0bcd8aaL586). 

The following patch gets us to:
```
diff --git a/pxr/imaging/hio/OpenEXR/openexr-c.c b/pxr/imaging/hio/OpenEXR/openexr-c.c
index 5c39b5adb..11464923d 100644
--- a/pxr/imaging/hio/OpenEXR/openexr-c.c
+++ b/pxr/imaging/hio/OpenEXR/openexr-c.c
@@ -657,7 +657,8 @@ static exr_result_t _nanoexr_rgba_decoding_initialize(
     for (int c = 0; c < decoder->channel_count; ++c) {
         int channelIndex = -1;
         decoder->channels[c].decode_to_ptr = NULL;
-        if (strIsComponent(layerName, decoder->channels[c].channel_name, "red")) {
+        if (strIsComponent(layerName, decoder->channels[c].channel_name, "red")
+            || strIsComponent(layerName, decoder->channels[c].channel_name, "y")) {
             rgba[0] = c;
             channelIndex = 0;
         }
```
![image](https://github.com/user-attachments/assets/09f29e10-bf63-4b74-bc6a-5d2e0cf3ab35)

So the only thing now missing is the expansion from "YA" to "YYYA" for the RG16F and RG32F images.

@meshula Would you be able to help with this?

### Steps to Reproduce
1. See #2800 for steps and files to use.

### System Information (OS, Hardware)
Windows: heap corruption crash before the scene shows
Linux: Corrupted images

### Package Versions
USD v24.08 with MaterialX v1.38.10

**Comment by jesschimein** [comment]
Filed as internal issue #USD-10112

**Comment by meshula** [comment]
Do you have a usd file and images corresponding to your test that you could attach? Perhaps whatever is driving your screenshot?

**Comment by seando-adsk** [comment]
Yes sure. In MayaUsd we have a unit test for this. It is part of our `testVP2RenderDelegateMaterialX` test, specifically the [DemoQuads](https://github.com/Autodesk/maya-usd/blob/dev/test/lib/mayaUsd/render/vp2RenderDelegate/testVP2RenderDelegateMaterialX.py#L168) test which is failing. The reference image is [DemoQuads_render.png](https://github.com/Autodesk/maya-usd/blob/dev/test/lib/mayaUsd/render/vp2RenderDelegate/VP2RenderDelegateMaterialXTest/baseline/DemoQuads_render.png).  The usd file it loads is [DemoQuads.usda](https://github.com/Autodesk/maya-usd/blob/dev/test/testSamples/MaterialX/DemoQuads.usda)

Sean

**Comment by sunyab** [comment]
Closing out old issue, this was addressed by #3343.
